### PR TITLE
feat(*): add algolia as the searcher

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
Algolia provides a _much_ better contextual + overall search than the builtin plugin. It works by crawling the site (which happens every day at 12:00 PM UTC, or can be manually triggered at any time) and then indexing the entire site. It supports CMD+K too!

Comparison of results:
Current
image

Algolia
image

## Related Issues
N/A

## Release Information
next

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

**Review by connorsmith256 — APPROVED** [review]
Wow, this is great!

**Comment by lachieh** [comment]
love it, do it, merge it

File: docusaurus.config.js [hunk 1]
```diff
@@ -164,6 +162,12 @@ const config = {
         ],
         copyright: `Copyright © ${new Date().getFullYear()} wasmCloud LLC. All rights reserved. The Linux Foundation has registered trademarks and uses trademarks. For a list of trademarks of The Linux Foundation, please see our Trademark Usage page: https://www.linuxfoundation.org/trademark-usage. Built with Docusaurus.`,
       },
+      algolia: {
+        apiKey: 'f0ef30f3d98ce5e9a7dd7579bb221dfc',
```
Comment (ricochet) [inline docusaurus.config.js:166]: should this be a secret?

Comment (lachieh) [reply docusaurus.config.js:166]: yes 😞

```suggestion
        apiKey: process.env.ALGOLIA_API_KEY,
```

I believe the `appId` and `indexName` are not secured but the apiKey is used to update the index during build

Comment (brooksmtownsend) [reply docusaurus.config.js:166]: re: https://docusaurus.io/docs/search#using-algolia-docsearch it's okay
```
      // The application ID provided by Algolia
      appId: 'YOUR_APP_ID',

      // Public API key: it is safe to commit it
      apiKey: 'YOUR_SEARCH_API_KEY',
```

Comment (brooksmtownsend) [reply docusaurus.config.js:166]: image
...

In our acceptance email they recommended copy/pasting this directly into the frontend as a script, maybe I misunderstand? I can regenerate that key and invalidate it if necessary :(

Comment (brooksmtownsend) [reply docusaurus.config.js:166]: As long as I didn't pick the wrong key (which pretty sure this is a search key) this is what Docusaurus does https://github.com/facebook/docusaurus/blob/main/website/docusaurus.config.js#L493

Comment (lachieh) [reply docusaurus.config.js:166]: Does algolia index the site from their side? how do the content changes get indexed on build?

Comment (brooksmtownsend) [reply docusaurus.config.js:166]: They don't get indexed on build, it actually gets indexed on a daily interval or manually. I need to check on getting that integrated into a pipeline though

Comment (lachieh) [reply docusaurus.config.js:166]: ok, cool, this still looks good then!

**Pull request merged**